cmd/cli/app/datasource/common.go

```diff
@@ -0,0 +1,148 @@
+// SPDX-FileCopyrightText: Copyright 2023 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datasource
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/viper"
+
+	"github.com/mindersec/minder/internal/util"
+	"github.com/mindersec/minder/internal/util/cli"
+	"github.com/mindersec/minder/internal/util/cli/table"
+	"github.com/mindersec/minder/internal/util/cli/table/layouts"
+	minderv1 "github.com/mindersec/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+// executeOnOneDataSource executes a function on a single data source
+func executeOnOneDataSource(
+	ctx context.Context,
+	t table.Table,
+	f string,
+	dashOpen io.Reader,
+	proj string,
+	exec func(context.Context, string, *minderv1.DataSource) (*minderv1.DataSource, error),
+) error {
+	ctx, cancel := cli.GetAppContext(ctx, viper.GetViper())
+	defer cancel()
+
+	reader, closer, err := util.OpenFileArg(f, dashOpen)
+	if err != nil {
+		return fmt.Errorf("error opening file arg: %w", err)
+	}
+	defer closer()
+
+	ds := &minderv1.DataSource{}
+	if err := minderv1.ParseResource(reader, ds); err != nil {
```

Comment:
We need to use `ParseResourceProto` instead. This won't work because of the `oneof` usage.